primitives/src/locktime/relative.rs

```diff
@@ -432,6 +431,46 @@ impl From<LockTime> for Sequence {
     fn from(lt: LockTime) -> Sequence { lt.to_sequence() }
 }
 
+#[cfg(feature = "serde")]
+impl serde::Serialize for LockTime {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_u32(self.to_consensus_u32())
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> serde::Deserialize<'de> for LockTime {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
```

Comment:
In 0aea1a03f1d49f651097a7fcadfa4398ae739298:

Why is this function implementation so big instead of symmetrically (to serialize) just calling `u32::deserialize(deserializer)?` then attempting to convert the result to a locktime?

Reply:
If this didn't work then the the implication is that serde doesn't work for `u32`. If true, I don't see why it'd be our problem.

Github has kinda hidden the CI failures but I think I found the failed run you are alluding to and it appears to be a compilation issue around a function called `visit_u32` which wouldn't be needed if we were just going through u32.